main.go

```diff
@@ -1,89 +1,51 @@
+	"github.com/afdesk/scan2html/render"
 )
 
 func main() {
-	if common.IsHelp() || len(os.Args) <= 1 {
-		helpMessage()
-	}
-	pluginFlags, trivyCommand := common.RetrievePluginArguments(availableFlags)
-	jsonResultFile := pluginFlags["--load-result"]
-	if jsonResultFile == "" {
-		jsonResultFile = filepath.Join(os.TempDir(), tempJsonFileName)
-		err := common.MakeTrivyJsonReport(trivyCommand, jsonResultFile)
-		if err != nil {
-			log.Fatalf("failed to build report: %v", err)
-		}
-		defer func(file string) {
-			if err := os.Remove(file); err != nil {
-				log.Printf("failed to remove file %v", err)
-			}
-		}(jsonResultFile)
-	}
-
-	firstHTML, err := common.ReadPluginFile("first.html")
+	rootCmd, err := initFlags()
 	if err != nil {
-		log.Fatalf("failed to read html file: %v", err)
+		log.Fatalf("%+v", err)
 	}
 
-	reportJson, err := os.ReadFile(jsonResultFile)
-	if err != nil {
-		log.Fatalf("failed to read json file: %v", err)
-	}
-
-	secondHTML, err := common.ReadPluginFile("second.html")
-	if err != nil {
-		log.Fatalf("failed to read html file: %v", err)
+	if err = rootCmd.Execute(); err != nil {
+		log.Fatalf("%+v", err)
 	}
+}
 
-	createdAt := time.Now().Unix()
-	argsStr := strings.Replace(strings.Join(os.Args[1:], " "), string(filepath.Separator), "/", -1)
-
-	output := bytes.NewBuffer(firstHTML)
-	fmt.Fprintf(output, "const trivyData = %s;\nconst createdAt = %d;\nconst args = %q;\n%s", reportJson, createdAt, argsStr, secondHTML)
+func initFlags() (*cobra.Command, error) {
+	var fileName string
+	rootCmd := &cobra.Command{
+		Use: "scan2html",
+		CompletionOptions: cobra.CompletionOptions{
+			DisableDefaultCmd: true,
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runPlugin(fileName)
+		},
+	}
+	rootCmd.PersistentFlags().StringVar(&fileName, "file", "trivy-report.html", "file to save html report")
+	return rootCmd, nil
+}
 
-	htmlResultOutput := pluginFlags["--html-result"]
-	if htmlResultOutput == "" {
-		log.Println("--html-result flag is not defined. Set default value result.html")
-		htmlResultOutput = "result.html"
-	}
-	err = os.WriteFile(htmlResultOutput, output.Bytes(), 0600)
+func runPlugin(fileName string) error {
+	inputData, err := io.ReadAll(os.Stdin)
 	if err != nil {
-		log.Fatalf("failed to write output file: %v", err)
+		return xerrors.Errorf("error reading trivy output: %v\n", err)
 	}
-}
-
-func helpMessage() {
-	_, err := fmt.Printf(`
-scan2html v%s
-Usage: trivy scan2html [-h,--help] command target filename
- A Trivy plugin that scans and output the results to a html file.
-Options:
-  -h, --help    Show usage.
-Examples:
-  # Scan 'alpine:latest' image
-  trivy scan2html image alpine:latest --html-result result.html
-  # Scan local folder
-  trivy scan2html fs . --html-result result.html
-`, version)
+	err = render.Render(fileName, inputData)
 	if err != nil {
-		log.Fatalf("Failed to display help message %v", err)
+		return err
```

Comment:
maybe we should add some comment for this error too? It will be hard to look for a reason of unknown errors